choreolib/src/main/java/choreo/auto/AutoRoutine.java

```diff
@@ -28,62 +29,38 @@ public class AutoRoutine {
    * The factory that created this loop. This is used to create commands that are associated with
    * this loop.
    */
-  protected final AutoFactory factory;
+  private final AutoFactory factory;
 
   /** The underlying {@link EventLoop} that triggers are bound to and polled */
-  protected final EventLoop loop;
+  private final EventLoop loop = new EventLoop();
 
   /** The name of the auto routine this loop is associated with */
-  protected final String name;
+  private final String name;
+
+  /** The alliance helper that is used to determine flipping logic */
+  private final AllianceContext allianceCtx;
 
   /** A boolean utilized in {@link #active()} to resolve trueness */
-  protected boolean isActive = false;
+  private boolean isActive = false;
 
   /** A boolean that is true when the loop is killed */
-  protected boolean isKilled = false;
+  boolean isKilled = false;
```

Comment:
Would it be useful to expose this to the user as a `killed()` method that returns a trigger?

Reply:
what loop would that be polled on? kinda an anit-pattern

Reply:
Didn't think of that, ignore this then